### ISSUE TITLE
should also parse __init__.py files

### ISSUE THREAD

**Issue by maharvey** [lucsorel/py2puml#52]
I am trying to analyze an inherited project, with a structure:
```
    package
        subpackage
            __init__.py     # base classes and interfaces
            classfile.py   # derived classes
```

If I run: `py2puml package/subpackage package.subpackage` it will find any classes defined in `classfile.py`, but will not list any classes from the `__init__.py`

Since the entire project is organized this way, and it's not my code, I am unable to use py2puml on it.

**Comment by lucsorel** [comment]
hi @maharvey :smiley: 

Thank you for your interest in this project and taking the time to report your issue.

**TLDR**

`__init__` is not a sub-module of `package.subpackage`, have you tried the command `py2puml package package` command? There, you should find all the classes of `subpackage/__init__.py`, along the ones of all modules and packages sibling to `subpackage`. Can you tell me what you get, please?

**Longer explanation**

The `py2puml package/subpackage package.subpackage` command asks `py2puml` to inspect the data structures (classes, enums, named tuples, dataclasses) of the modules that are children of the `package.subpackage` package. The `__init__.py` file is not a sub-module of `subpackage`, it is the content of the `subpackage` module itself, which is also a package because it is the name of the folder.

If you want to define classes in the `subpackage` module, you have 2 options :

1. define them in a `subpackage.py` file, but then it is not possible to have sub-modules of `subpackage.py` (because it is a file and not folder)

```text
package
    subpackage.py
        subsubmodule.py cannot be a "children" of subpackage.py on the disk
```

2. create a `subpackage` folder with a `__init__.py` file in it, in which you define classes, create as many sub-module files and sub-packages as you need

```text
package
    subpackage
        __init__.py  # define a Foo class here for instance
        subsubmodule.py  # from package.subpackage import Foo
```

To import the Foo class in the `package.subpackage.subsubmodule` module, you would type:

```python
from package.subpackage import Foo
```

And not:

```python
from package.subpackage.__init__ import Foo
```

However, writing all this, it seems to me that one would expect the `py2puml package/subpackage package.subpackage` command to yield the data structures of `package.subpackage` and not only its children ones.

Can you tell me if you see the classes defined in `subpackage/__init__.py` when you run the  `py2puml package package`  command, please? It will also yield the classes in other subpackages sibling to `package/subpackage` and I understand it is not what you wanted :smiley:

**Comment by JustKiddingCode** [comment]
I am running into the same problem, one level higher.
If I define my classes directly into the `package/__init__.py` file, there is no possibility to create the plantuml files.

**Comment by lucsorel** [comment]
Thank you @JustKiddingCode for reporting the issue too. Given a package or a module (the one passed as an argument to the CLI), the tool looks for sub-modules or sub-packages without looking for the given module itself.

This is fixable, I just lack the time; I am sorry, I cannot provide a deadline for this.
